rotkehlchen/tests/unit/accounting/test_exchanges.py

```diff
@@ -38,3 +42,75 @@ def test_account_for_coinbase_ledger_actions(rotkehlchen_api_server_with_exchang
     assert events_map[AccountingEventType.FEE] == 2
     assert events_map[AccountingEventType.ASSET_MOVEMENT] == 3
     assert sum(events_map.values()) == 11  # processed events
+
+
+@pytest.mark.parametrize('default_mock_price_value', [FVal(1.5)])
+@pytest.mark.parametrize('ethereum_accounts', [[]])
+def test_exchanges_removed_api_keys(rotkehlchen_api_server):
+    """
+    Test that if actions made on an exchange are stored in the DB but the API keys were removed
+    then the actions are still taken into account in accounting.
+    """
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    with rotki.data.db.user_write() as write_cursor:
+        rotki.data.db.add_trades(write_cursor, trades=[Trade(
+            timestamp=Timestamp(1611426201),
+            amount=ONE,
+            base_asset=A_ETH,
+            quote_asset=A_BTC,
+            trade_type=TradeType.BUY,
+            rate=FVal(1.5),
+            location=Location.COINBASE,
+        ), Trade(
+            timestamp=Timestamp(1611426201),
+            amount=FVal(7),
+            base_asset=A_USDT,
+            quote_asset=A_LINK,
+            trade_type=TradeType.SELL,
+            rate=FVal(7),
+            location=Location.EXTERNAL,
+        )])
+        rotki.data.db.add_asset_movements(write_cursor, asset_movements=[AssetMovement(
+            timestamp=Timestamp(1611426201),
+            location=Location.COINBASE,
+            category=AssetMovementCategory.DEPOSIT,
+            asset=A_BTC,
+            amount=ONE,
+            fee_asset=A_BTC,
+            fee=FVal(0.00001),
+            address=None,
+            transaction_id=None,
+            link='no link',
+        )])
```

Comment:
also add a ledger action and a margin position to cover the entire array of events that an exchange can add as of this version of rotki